store/sqlstore/channel_store.go

```diff
@@ -1478,6 +1478,53 @@ func (s SqlChannelStore) RemoveMember(channelId string, userId string) store.Sto
 	})
 }
 
+func (s SqlChannelStore) RemoveAllDeactivatedMembers(channelId string) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		query := ""
+		if s.DriverName() == model.DATABASE_DRIVER_POSTGRES {
+			query = `
```

Comment:
2/5, but perhaps we can avoid the need to work around MySQL's limitations here by just forming the `UserId` subquery of all users that have been deleted, regardless whether they are actually `ChannelMembers` of the channel in question. The existing parent constraint of `ChannelMembers.ChannelId` kicks in anyway.

Reply:
I'll have another try with some different queries and see how it goes.